result/array-bracket-spacing.js

```diff
@@ -0,0 +1,118 @@
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var [ x, y ] = z;
+
+var arr = ['foo', 'bar'];
+var [x,y] = z;
+/*eslint array-bracket-spacing: ["error", "never"]*/
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var arr = ['foo', 'bar' ];
+var arr = [ ['foo'], 'bar'];
```

Comment:
__[development][eslint]__
* Error - 'arr' is already defined (no-redeclare)
* Error - There should be no space after '[' (array-bracket-spacing)
* Error - A space is required after '[' (array-bracket-spacing)
* Error - A space is required before ']' (array-bracket-spacing)